internal/configuration.go

```diff
@@ -0,0 +1,17 @@
+package internal
+
+import "github.com/caarlos0/env/v11"
+
+type Configuration struct {
```

Comment:
The code currently depends directly on environment variable parsing, which can introduce issues if the keys are not set. Consider adding validation on keys along with detailed error messages to notify which variable is missing.